### DISCUSSION TITLE
Network issues (?) when installing Talos on a singlenode x86_64 machine?

### DISCUSSION THREAD

**Discussion by kastl-ars** [siderolabs/talos#9046] [Q&A] [answered]
Hi all,

how can I make sure that a baremetal node has a working network connection after booting up from the ISO?

I am trying to deploy Talos to a singlenode x86_64 device. I found the `ip=` kernel parameter, so I can set IP address, gateway, netmask etc. pp.

- I can ping the node from another network and the router in this network, so it seems reachable and the gateway seems to work
- The node always shows an error when trying to sync the time. Either it reports that it cannot reach the `time.cloudflare.com` servers via UDP port 123. Or it seems it cannot resolve this hostname using the DNS servers I specify.
- Using a custom NTP server via the `ip=` parameter seems to also fail.

Unfortunately I cannot log in (as intended).

Is there any kernel parameter that sets a more verbose logging?
Is there a talosctl command that tries to reach the machine and checks things (like a pre-flight check)?

Just applying the configuration (`talosctl apply-config ...`) does not seem to do anything, either because it is stuck downloading things due to bad/missing internet or for other reasons. It does not output anything on the terminal/screen so no hint at what is happening.

Using another machine with the same network cable, setting the IP address etc. manually and trying to resolve DNS or NTP I do not see any errors.

Any ideas how to debug this?

Thanks in advance!

Kind Regards,
Johannes

**Comment by kastl-ars** [comment]
I just tried again, the ping stops working sometime after the `apply-config` step. But the machine's terminal does not show anything and has been sitting idle for 20 minutes already.

**Reply by smira** [reply, marked as answer]
The only way to debug Talos while networking is not set up is to use the console.

The best way is to hook up to the serial console (physical or virtual via IPMI or similar).

If that is not available, use video console. You can switch from the dashboard view to the Linux console via `Alt` + `F1` to see logs only. 

Simple "ping" would be `talosctl disks --insecure -n <IP>` before the machine config is applied.

You can also inspect the current state via resources: `talosctl get addresses --insecure -n <IP>`, etc.

If you use static network configuration via kernel cmdline `ip=`, make sure you apply the same via machine config as well.

**Reply by kastl-ars** [reply]
Thanks for the fast reply.

I could successfully ping the machine and `talosctl disks --insecure -n <IP>` showed the disks the machine has. So I daresay the configuration should be fine. Maybe the errors came from during the boot, before the network was setup completely?

I'll give it another try and check the addresses as you mentioned.

Is there a command to follow along after applying the configuration? So I could find out why the ping suddenly stops?

**Reply by smira** [reply]
Please see my previous message:

> If you use static network configuration via kernel cmdline ip=, make sure you apply the same via machine config as well.

On reboot/install Talos will lose `ip=` arg, so you'd better setup networking in the machine configuration.

**Reply by kastl-ars** [reply]
> On reboot/install Talos will lose `ip=` arg, so you'd better setup networking in the machine configuration.

I have that set up, that does not seem to be the problem.

Right now I cannot get the machine to be pingable, either the boot errors out due to EFI STUB errors (caused by adding only the `ip=` argument in GRUB) or it boots but is not pingable.

This also happen with an older ISO (1.7.0).

I'll check if the hardware is broken to make sure I am not hunting ghosts here...

**Reply by kastl-ars** [reply]
OK, the hardware is working, I could successfully install another linux distro using the network.

I tried again, this time also setting `ipv6.disable=1` as my network does not support it (thanks, ISP...).

This time it shows the same errors during boot, but it is pingable, the apply-config returns something about "applied without reboot". And I could successfully continue with bootstrapping, fetching the kubeconfig and getting the nodes.

The `talosctl get addresses...` command also showed me the addresses, but without much details. But at least something seems to be reachable.

So I would say I know how to proceed. Thanks, @smira for your help!
